mono/mini/interp/interp.c

```diff
@@ -4519,6 +4519,7 @@ interp_exec_method_full (InterpFrame *frame, ThreadContext *context, FrameClause
 						THROW_EX (mono_error_convert_to_exception (error), ip);
 					frame_objref (frame) = mono_object_new_checked (imethod->domain, newobj_class, error);
 					mono_error_cleanup (error); /* FIXME: don't swallow the error */
+					error_init_reuse (error);
```

Comment:
IMO the right approach is to use a local `MonoError` for the two calls in this branch that need to ignore the error.

The lifecycle of a `MonoError` should be simple.  `error_init_reuse` is a code smell.

Reply:
We are super concerned with stack usage here.
This is an intense loop and reuse is appropriate.

Reply:
The interpreter doesn't recurse here.  But it does make a call to another function already.  It's perfectly reasonable to call a `_nofail` version instead of `_checked` one here.

Reply:
Some of these are in `interp_exec_method_full`, some are not.
`ERROR_DECL` must not be used in `interp_exec_method_full` (not even in rare/slow paths), but it can be used in non-inlined functions that it calls.
So yeah we'd have to introduce a few helper functions.
I suggest `mono_interp_foo` and not `mono_foo_nofail`, as these are also anti-patterns -- ignoring errors, don't want to bless it.
I still think reuse is ok, but non-reuse should be viable here.

Reply:
I can do it.

Reply:
I still think reuse is the way.
The interpreter went out of its way already to reuse, just was slightly off.
The interpreter I think is unusual in its quest for performance since it really wants to be a JIT or AOT.
I worry that exiting to runtime& recursing
back to interp will double error use, though maybe never deeply?
Contructors bottom out quickly?